### DISCUSSION TITLE
Getting Started with Wi-Fi on ESP-IDF · Espressif Developer Portal

### DISCUSSION THREAD

**Discussion by giscus** [espressif/developer-portal#214] [General]
# Getting Started with Wi-Fi on ESP-IDF · Espressif Developer Portal

The developer resources in just one place!

https://developer.espressif.com/blog/getting-started-with-wifi-on-esp-idf/

**Comment by auntybellum** [comment]
Hello, I very much appreciated the work you put into this. I am quite good at connecting WiFI by both methods and mqtt as well. I would love a more advanced tutorial as I am struggling to understand the relationships between esp/event loops, freertos/ event loops and how rtos tasks and the opportunities or Core 1 and Core 2 could be exploited to simplify code.
Thank you again

**Comment by Swarkin** [comment]
The call to `esp_wifi_set_default_wifi_sta_handlers()` seems to be unnecessary since `esp_netif_create_default_wifi_sta()` already calls it internally.
Furthermore, `esp_wifi_set_mode(WIFI_MODE_STA)` is unnecessary when the program does not use AP functionality since station is already the default wifi mode.

**Comment by dinosaurclover-dev** [comment]
I thought I had struck gold when I found this. I need to be able to switch off the wifi and switch it back on again later. I simply modified wifi_tutorial.c by repeating the code twice. Here's what I did in simplified form.

tutorial_init(); tutorial_connect(); tutorial_disconnect(); tutorial_deinit();
tutorial_init(); tutorial_connect(); tutorial_disconnect(); tutorial_deinit();
Does not work.

I also tried
tutorial_init(); tutorial_connect(); tutorial_disconnect();tutorial_connect(); tutorial_disconnect(); tutorial_deinit();
Second connect does not work.

Any thoughts?
